src/core/ad.jl

```diff
@@ -198,3 +198,7 @@ for F in (:link, :invlink)
         end
     end
 end
+
+# This function will call emptyrdcache only if ReverseDiff & Memoization are loaded
+# See Turing issue #1418
+clearcache(x=nothing) = ()
```

Comment:
why do we need an argument here?

Reply:
This was just a stupid trick to get the compiler to dispatch on the version I overwrite it with when we load ReverseDiff and Memoization. There's probably a better way to do this.

Reply:
In general, I don't think it's good to overwrite method definitions. IMO `clearcache` is not needed, instead one could add a check before calling `emptyrdcache()` such as
```julia
if isdefined(Turing, :emptyrdcache)
    emptyrdcache()
end
```